docs/modules/ROOT/pages/jar-lib.adoc

```diff
@@ -0,0 +1,59 @@
+# Using DataSonnet Programmatically
+:toc:
+:toclevels: 1
+
+DataSonnet is distributed as a JAR which contains all necessary classes and dependencies to use DataSonnet programmatically embedded in your projects. Use the following Maven snippet to add DataSOnnet as a dependency to your `pom.xml`:
+
+[source,xml]
+----------
+<dependency>
+    <groupId>com.datasonnet</groupId>
+    <artifactId>datasonnet-mapper</artifactId>
+    <version>${datasonnet.version}</version>
+</dependency>
+----------
+
+where `${datasonnet.version}` is the latest released version (currently 1.0.4).
+
+The simplest scenario is mapping from JSON to JSON with no additional variables:
+
+[source,java]
+------
+String json = ...; //JSON object as string
+
+Mapper mapper = new Mapper(datasonnetScript);
+String transformedJson = mapper.transform(json)
+------
+
+More complex scenarios may involve passing additional variables as well as xref:dataformats.adoc[transforming to/from other supported data formats]:
+
+[source,java]
+------
+String jsonData = ...;
+
+String jsonVar = ...;
+String xmlVar = ...;
+String textVar = ...;
+
+Map<String, Document> variables = new HashMap<>();
+variables.put("varJson", new StringDocument(jsonVar, "application/json"));
+variables.put("varXml", new StringDocument(xmlVar, "application/xml"));
+variables.put("varTxt", new StringDocument(textVar, "text/plain"));
+
+Mapper mapper = new Mapper(datasonnetScript, variables.keySet());
+StringDocument transformedResult = mapper.transform(new StringDocument(jsonData, "application/json"), variables, "application/json");
```

Comment:
this doesn't return a StringDocument, just a Document

Reply:
fixed